File: frontend/src/types/variety.ts
export interface VarietyData {
  name: string;
  price: number;
}

export interface VarietyItemRef {
  id: string;
  name: string;
}

export interface Variety {
  id: string;
  name: string;
  inventoryItemIds: string[];
  inventoryItems: VarietyItemRef[];
  varieties: VarietyData[];
  ownerId: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateVarietyRequest {
  name: string;
  inventoryItemIds: string[];
  varieties: VarietyData[];
  ownerId: string;
}

export interface UpdateVarietyRequest {
  name?: string;
  inventoryItemIds?: string[];
  varieties?: VarietyData[];
}

File: src/Inventory.API/Data/Entities/Variety.cs
namespace Inventory.API.Data.Entities;

public sealed class Variety
{
    public string Id { get; set; } = "";
    public string Name { get; set; } = "";
    public string InventoryItemIds { get; set; } = "[]";
    public List<VarietyData> Varieties { get; set; } = new();
    public string OwnerId { get; set; } = "";
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }
}

public sealed class VarietyData
{
    public string Name { get; set; } = "";
    public decimal Price { get; set; }
}
